packages/docusaurus-plugin-content-docs/src/docFrontMatter.ts

```diff
@@ -14,14 +14,29 @@ import {
 } from '@docusaurus/utils-validation';
 import {DocFrontMatter} from './types';
 
+// NOTE: This creates a validator from given joi type,
+// that allows the object with any fields, as long as they are
+// of the given type, or just a raw type given
+
+const platformDependant = (joiType: any): any =>
```

Comment:
Type should be `Joi.AnySchema` although I wonder if it can't be narrowed to just boolean/number/string

Reply:
In case we would like to have something like screen size dependent logos etc, the joiType can be more complex object type, it would be also good to probably run validation of it in .desktop etc options in config

Reply:
`Joi.AnySchema` is the widest schema type possible. Also, the `key` is totally useless because the same config would pass with them or not, unless you have different value schema for `desktop` / `mobile` than the other cases, which you don't